### PR TITLE
Fix image to text node, it was bugged 

### PR DESCRIPTION
**Pull request by VinciGit00**

### CODE REVIEW

File: scrapegraphai/graphs/smart_scraper_graph.py [hunk 1]
```diff
@@ -71,8 +71,7 @@ def _create_llm(self, llm_config: dict):
             return OpenAI(llm_params)
         elif "gemini" in llm_params["model"]:
             return Gemini(llm_params)
-        else:
-            raise ValueError("Model not supported")
+        raise ValueError("Model not supported")
```
Comment (PeriniM) [inline scrapegraphai/graphs/smart_scraper_graph.py:74]: there is an indentation error

File: scrapegraphai/nodes/image_to_text_node.py [hunk 2]
```diff
@@ -10,34 +10,39 @@ class ImageToTextNode(BaseNode):
...
-        llm (OpenAIImageToText): An instance of the OpenAIImageToText class.
+        llm_model (OpenAIImageToText): An instance of the OpenAIImageToText class.
 
     Methods:
         execute(state, url): Execute the node's logic and return the updated state.
     """
 
-    def __init__(self, llm, node_name: str):
+    def __init__(self, input: str, output: List[str], model_config: dict,
+                 node_name: str = "GetProbableTags"):
```
Comment (PeriniM) [inline scrapegraphai/nodes/image_to_text_node.py:20]: change default node_name

File: scrapegraphai/nodes/image_to_text_node.py [hunk 3]
```diff
@@ -10,34 +10,39 @@ class ImageToTextNode(BaseNode):
...
-            llm (OpenAIImageToText): An instance of the OpenAIImageToText class.
-            node_name (str): name of the node
+            input (str): The input for the node.
+            output (List[str]): The output of the node.
+            model_config (dict): Configuration for the model.
+            node_name (str): Name of the node.
         """
-        super().__init__(node_name, "node")
-        self.llm = llm
+        super().__init__(node_name, "node", input, output, 2, model_config)
```
Comment (PeriniM) [inline scrapegraphai/nodes/image_to_text_node.py:30]: should accept only 1 input which is the url or list of url present in the state

File: scrapegraphai/nodes/image_to_text_node.py [hunk 4]
```diff
@@ -10,34 +10,39 @@ class ImageToTextNode(BaseNode):
...
-        :return: The updated state after executing this node.
-        """
+            url (str): URL of the image to process.
 
+        Returns:
+            dict: The updated state after executing this node.
+        """
         print("---GENERATING TEXT FROM IMAGE---")
-        text_answer = self.llm.run(url)
+        text_answer = self.llm_model.run(url)
```
Comment (PeriniM) [inline scrapegraphai/nodes/image_to_text_node.py:45]: missing the retrieval part from the state before this line to get the url

File: scrapegraphai/nodes/image_to_text_node.py (hunk 2 above)
Comment (VinciGit00) [reply scrapegraphai/nodes/image_to_text_node.py:20]: did it

File: scrapegraphai/nodes/image_to_text_node.py (hunk 3 above)
Comment (VinciGit00) [reply scrapegraphai/nodes/image_to_text_node.py:30]: did it

File: scrapegraphai/nodes/image_to_text_node.py [hunk 5]
```diff
@@ -10,34 +10,39 @@ class ImageToTextNode(BaseNode):
...
+            output (List[str]): The output of the node.
+            model_config (dict): Configuration for the model.
+            node_name (str): Name of the node.
         """
-        super().__init__(node_name, "node")
-        self.llm = llm
+        super().__init__(node_name, "node", input, output, 2, model_config)
+        self.llm_model = model_config["llm_model"]
 
     def execute(self, state: dict, url: str) -> dict:
```
[the remark this answers is not in the document]
Comment (VinciGit00) [reply scrapegraphai/nodes/image_to_text_node.py:33]: you are right

**Review by PeriniM — APPROVED** [review]

**Pull request merged**